### PR TITLE
Core/python process handler

### PR DESCRIPTION
**Pull request by roigcarlo**
This changes the handler of process from unique_ptr to shared_ptr. 
This allows to:
- Is possible to accept Process::Pointer a parameter in functions exposed to python
- Process are handled now like they were handled with boost.python.

@KratosMultiphysics/team-maintainers Please take 5 minutes to check codes run and review your application specific codes. Please take special care to review processes that we not directly derived from the class 'Process' since those were trickier to find and I may have missed some.

In the process of making the PR I noticed that some process are not exporting the Kratos pointer. I will note them here so you can handle this situation as you prefer (either remove them from the python interface or add the definition). The application were not ported anyway so there will be no problems compiling right now:

**Incompressible**:
- SaveFlagModelPartProcess
- SaveElementBySizeProcess
- SaveElementByFlagProcess
- ChooseElementProcess
- AssignHByDistanceProcess

**LagrangianMPM**:
- RecomputeNeighboursProcess
- Gauss_Coordinates_Update_Process (also the name of the class should be camelcase :D)
- ComputeMLSShapeFunctionsProcess
- ComputeLMEShapeFunctionsProcess

### CODE REVIEW

**Review by jcotela — APPROVED** [review]
:+1: Thanks for taking care of this.

Regarding applications, I checked specifically the fluid, thermo mechanical and HDF5. I don't expect problems with those.

**Comment by ipouplana** [comment]
I checked Dam and Poromechanics Applications and it seems right to me. Thanks!

**Comment by roigcarlo** [comment]
I am holding this for #2126 to avoid conflicts, I will merge at about 5:00 if everything its ok

**Pull request merged**
